### PR TITLE
JBPM-4833 - Data Modeller MVP, Tests, and related improvments

### PR DESCRIPTION
**Pull request by wmedvede**
The following commit includes MPV improvments, tests, and small fixes for the following components.

	- MainDataObjectEditor
	- MainDataObjectFieldEditor
	- PackageSelector
	- NewPackagePopup
	- DroolsDataObjectEditor
	- DroolsDataObjectFieldEditor
	- JPADataObjectEditor
	- JPADataObjectFieldEditor
	- AdvancedDataObjectEditor
	- AdvancedDataObjectFieldEditor

### CODE REVIEW

**Comment by wmedvede** [comment]
The following PR https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/109 should be merged into master before this one.

**Comment by wmedvede** [comment]
Hello @manstis 
would you mind review this PR?

**Comment by wmedvede** [comment]
cherry-pick into 6.3.x DO NOT APPLY for this PR

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/handlers/jpadomain/command/AdjustFieldDefaultRelationsCommand.java [hunk 1]
```diff
@@ -87,7 +89,7 @@ public void execute() {
             if ( field.isMultiple() ) {
                 if ( field.getAnnotation( JPADomainAnnotations.JAVAX_PERSISTENCE_ELEMENT_COLLECTION ) == null ) {
                     List<ValuePair> values = new ArrayList<ValuePair>(  );
-                    values.add( new ValuePair( "fetch", "EAGER" ) );
+                    values.add( new ValuePair( RelationshipAnnotationValueHandler.FETCH, defaultFetchMode ) );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/handlers/jpadomain/command/AdjustFieldDefaultRelationsCommand.java]: Any reason you choose to pass the enum as the first parameter but the enum's name as the second? Wouldn't enums in both cases be better? Unless you have a very good reason (and if it's that good a reason, why not add some JavaDoc explaining the very good reason so (1) you don't forget in the future (2) if others edit this code in the future they know the very good reason too!

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java [hunk 2]
```diff
@@ -357,7 +357,85 @@
 
     String advanced_domain_value_pair_editor_popup_title();
 
-    //persistence editor i18n
+    //persistence domain editor
+
+    String persistence_domain_objectEditor_entity_category();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java]: Liking the i18n names :) -- shame the file is called ```Constants``` and not something like ```DataModellerConstants``` ;)

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/util/UIUtil.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+     * @param options List of pairs <"text to show", "value"> representing the select options.
+     *
+     * @param selectedValue The current selected value.
+     *
+     * @param includeEmptyItem true if the empty optio should be included.
+     *
+     *
+     */
+    public static void initList( final Select select,
+                                final List<Pair<String, String>> options,
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/util/UIUtil.java]: Code formatting.. but nothing to worry about.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/common/domain/FieldEditor.java [hunk 4]
```diff
@@ -42,6 +43,12 @@ public ObjectProperty getObjectField() {
         return objectField;
     }
 
+    public FieldEditor( DomainHandlerRegistry handlerRegistry,
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/common/domain/FieldEditor.java]: FYI.. code formatting

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/handlers/jpadomain/command/AdjustFieldDefaultRelationsCommand.java (hunk 1 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/handlers/jpadomain/command/AdjustFieldDefaultRelationsCommand.java]: The first param is not an enum, RelationshipAnnotationValueHandler.FETCH is a constant.
and the enum name, just gives me the constant "EAGER". So I really don't see any problems with this...

Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/handlers/jpadomain/command/AdjustFieldDefaultRelationsCommand.java]: @wmedvede Me neither; I thought the first parameter was an enum.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java (hunk 2 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java]: Assuming we'll have global i18 file per gwt client module, I think that to name it just "Constants" it's not a bad approach. I this way we have standardized way of naming them.
If you go for example to the guided-rule-editor-client module, there you will find a Constants.properties. 
But it's true that we are not using any standard way for naming i18n files, and i18n properties...

**Comment by manstis** [comment]
@wmedvede Nice re-factoring to MVP :)

Whilst I doubt the test coverage is great there are tests and the improved separation will help further testing as and when required. Good job.

+1 to merge.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java (hunk 2 above)
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/resources/i18n/Constants.java]: There are discussions with the Zanata team that they find it really difficult to find "source" documents in Zanata when trying to fix i81n issues. They have requested we move all i18n to a single module shared across screens. It's under discussion. 

See https://translate.jboss.org/webtrans/translate?project=drools-wb&iteration=6.4.0&localeId=ja&locale=en%20#doc:org.drools/drools-wb-guided-dtable-editor-client/org/drools/workbench/screens/guided/dtable/client/resources/i18n/AnalysisConstants. We have been trying to make the "source" files easier to find by giving them better names (Guided Rule Editor is one we've not changed!!)

So anything we can do to make their life easier the better :)

**Comment by csadilek** [comment]
Thanks!

master: 22f074d59e407f19c5e74677dc93fd8131884cb6

**Comment by wmedvede** [comment]
@csadilek Don't know why this PR says "Closed with unmerged commits" but my commits seems to be in master.

@psiroky  the collateral effect seems to be that the JIRA wasn't automatically updated with the commit info.
I've add this comments just in case there were any error, but the commits are now in master.

**Comment by csadilek** [comment]
@wmedvede that's just because the commits get cherry-picked, not merged and therefore end up with a different hash. We do this to avoid merge commits and keep a linear history.

**Comment by wmedvede** [comment]
@csadilek this should explain why the JIRA wasn't automatically update.
@psiroky ^^

**Pull request closed without merging**